bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -133,13 +274,19 @@ void shutdown() {
         LOG.info("Shutting down BookieNettyServer");
         isRunning.set(false);
         allChannels.close().awaitUninterruptibly();
-        for (ChannelManager channel : channels) {
-            channel.close();
+
+        if (eventLoopGroup != null) {
+            eventLoopGroup.shutdownGracefully();
         }
+        if (jvmEventLoopGroup != null) {
+            jvmEventLoopGroup.shutdownGracefully();
+            LocalBookiesRegistry.registerLocalBookieAddress(bookieAddress);
```

Comment:
Shoudn't this be unregister?  and as a nit I would expect that we would want to unregister first and then shutdown afterwards.